src/ReadConfig.cxx

```diff
@@ -231,10 +234,128 @@ load_unsigned(const IniFile &file, const char *name, unsigned *value_r)
 	return true;
 }
 
+static std::shared_ptr<IgnoreList>
+load_ignore_list(const std::string& path, Config::IgnoreListMap& ignore_lists) {
+
+	{
+		auto status = std::filesystem::status(path);
+
+		if (status.type() == std::filesystem::file_type::not_found) {
+			throw FormatRuntimeError("Cannot load ignore file: '%s' does not exist", path.c_str());
+		}
+
+		if (status.type() == std::filesystem::file_type::directory) {
+			throw FormatRuntimeError("Cannot load ignore file: '%s' is a directory", path.c_str());
+		}
+	}
+
+	std::ifstream file(path);
+	if (!file.is_open()) {
+		throw FormatRuntimeError("Cannot load ignore file: cannot open '%s' for reading", path.c_str());
+	}
+
+	auto ignore_list = std::make_shared<IgnoreList>();
+
+	{
+		std::string line;
+		size_t line_num = 0;
+		while (std::getline(file, line)) {
+			line_num++;
+			if (line.empty()) {
+				continue;
+			}
+
+			IgnoreListEntry entry{};
+
+			// Format:
+			//   artist;album;title
+			// Empty string means "ignore", so e.g.:
+			//   ;;title               Match any track titled "title".
+			//   artist                Match any track with artist "artist".
+			//   artist;;              Identical to the above.
+			//   artist;;title         Match any track titled "title" from artist "artist".
+			//   artist;album;title    Match everything.
+			//   ;;                    Ignored, empty entries will not be pushed.
+			//
+			// The format is designed easy to parse and unlikely to conflict with the metadata
+			// we see in the wild. Realistically, the only conflict you may get is a semicolon in
+			// the metadata, which you can escape with a single backslash, e.g.:
+			//   artist1\;artist2;;   Match artist "artist1;artist2"
+			//
+			// Note that the parser does not have entirely typical support for escaping:
+			// Escaping rules are only applied for the sequence immediately before a semicolon.
+			// Backslashes do not have any special meaning outside this context, so double backslashes
+			// are not collapsed anywhere else.
+			// When interpreted as an escape, backslashes are collapsed as you would expect.
+			//   artist1\\artist2;album         Match "artist1\\artist2" and album "album"
+			//   artist\\\\;album;              Match artist "artist\\" and album "album"
+			//   artist1\\\\\;artist2;album;    Match artist "artist1\\;artist2" and album "album"
+			//
+
+			std::vector<std::string> tokens;
+			std::string temp_token;
+			for (auto part : std::ranges::views::split(line, ';')) {
+				// max 3 tokens (artist, album, title), fail if we already have 3
+				if (tokens.size() == 3) {
+					throw FormatRuntimeError("Invalid ignore specification (too many tokens), line: %d", line_num);
+				}
+
+				std::string token = temp_token + std::string(part.begin(), part.end());
```

Comment:
The temporary `std::string` constructed here is pointless overhead

Reply:
Right, will fix.